hw/ip/otbn/dv/otbnsim/sim/insn.py

```diff
@@ -418,13 +429,13 @@ def execute(self, state: OTBNState) -> None:
 
         mul_res = a_qw * b_qw
 
-        acc = int(state.single_regs['acc'])
+        acc = state.wsrs.ACC.read_signed()
```

Comment:
As I said in the other comment ACC is always a positive number, so shouldn't we be using `read_unsigned` here and `write_unsigned` below and similar in the other MULQACC instructions?

Reply:
You're completely right. This PR didn't change behaviour (casting a `Register` to int is the same as `read_signed`). I've fixed this, along with some other similar errors as part of rewriting things in PR #3589.